apps/server/src/modules/files/repo/sync-files.repo.ts

```diff
@@ -0,0 +1,98 @@
+import { EntityManager, ObjectId } from '@mikro-orm/mongodb';
+import { Injectable } from '@nestjs/common';
+import { EntityId, FileRecordParentType, Task } from '@shared/domain';
+import { SyncFileItemMapper } from '../mapper';
+import { SyncFileItem, SyncFileItemData } from '../types';
+
+// This repo is used for syncing the new filerecords collection with the old files collection.
+// It can be removed after transitioning file-handling to the new files-storage-microservice is completed.
+@Injectable()
+export class SyncFilesRepo {
+	constructor(protected readonly _em: EntityManager) {}
+
+	async findTaskFilesToSync(batchSize = 50): Promise<SyncFileItem[]> {
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+		const itemDataList: SyncFileItemData[] = await this._em.aggregate(Task, [
```

Comment:
DeletedAt can we copied, but we should not touched binary files that are marked as scan status blocked.